PSSM_score_Peptide/SEQ.cpp

```diff
@@ -5,7 +5,6 @@
 
 void SEQ::setCopyNmber()
 {
-	_CopyNumber = 1.0;
```

Comment:
Why remove the line, if input is fasta without repeat or RPM this will fail (number of copies will be 0)

Reply:
We set the _CopyNumber to 1 in the constructor or to number from RPM (it is parameter of the constructor ):

`SEQ::SEQ(string & SeqString, const string & SeqName, const double CopyNumber, alphabet& alph) :  _Seq_Name(SeqName), _CopyNumber(CopyNumber), _alph(alph) {
	for (size_t k = 0; k < SeqString.length(); ++k) {
		char i = SeqString.at(k);
		_Seq.push_back(alph._alphabetMap[i]);
	}
	setSeq_Type();
	setCopyNmber();
};
`
if we keep the line it overrides the _CopyNumber. I test the code with the RPM file and regular faa file and it is works good.

Reply:
Maybe we should call setCopyNumber only when needed and not always

Reply:
We use `void SEQ::setCopyNmber()` in two options -  1. calculate pval scanning. 2. calculate shuffle scanning - > but we don't use this code anymore because we use hits_pcc instead. 

when I run the calculate pval scanning it is not going inside the `if (regex_search(_Seq_Name, match, rgx))` 
so I don't see why we need this function. we set the number of copies in the constructor.

Reply:
If we don't need the function at all then remove it BUT it is necessary for other operations such as cutoffs, we don't want to break existing interfaces and find out the damage later.

To resolve, add a boolean parameter of `isSetCopyNumber` to constructor with default of true.
only if `isSetCopyNumber` is true then call `setCopyNumber` function.
When not required set the parameter to false